src/request/request.rs

```diff
@@ -73,6 +73,28 @@ impl<State> Request<State> {
 
     /// Returns a convenient wrapper around an optional references to the query
     /// parameters in the request's uri with the provided `name`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use via::{Next, Request, Result};
+    ///
+    /// async fn hello(request: Request, _: Next) -> Result<String> {
+    ///     // Attempt to parse the first query parameter named `n` to a `usize`
+    ///     // no greater than 1000. If the query parameter doesn't exist or
+    ///     // can't be converted to a `usize`, default to 1.
+    ///     let n = request.query("n").first().parse().unwrap_or(1).min(1000);
+    ///
+    ///     // Get a reference to the path parameter `name` from the request uri.
+    ///     let name = request.param("name").required()?;
+    ///
+    ///     // Create a greeting message with the provided `name`.
+    ///     let message = format!("Hello, {}!\n", name);
+    ///
+    ///     // Send a response with our greeting message, repeated `n` times.
+    ///     Ok(message.repeat(n))
+    /// }
+    /// ```
     pub fn query<'a>(&self, name: &'a str) -> QueryParam<'_, 'a> {
```

Comment:
I like the way this ended up. It may seem a bit silly to wrap `Option` and `Result` the way that we do but it feels really nice to use. Being able to validate the presence of, parse, and then limit a numerical value in query params on 1 line without having to write a utility function that only used 1 or 2 times is a great developer experience. Here is an example of how I modified the hello function in the hello-world example to test everything out.

```rust
use via::{Next, Request, Result};

async fn hello(request: Request, _: Next) -> Result<String> {
    // Attempt to parse the query parameter `n` to a `usize` no greater than
    // 1000. If the query parameter doesn't exist or can't be converted to a
    // `usize`, default to 1.
    let n = request.query("n").first().parse().unwrap_or(1).min(1000);

    // Get a reference to the path parameter `name` from the request uri.
    let name = request.param("name").required()?;

    // Create a greeting message with the provided name.
    let message = format!("Hello, {}!\n", name);

    // Print a message to the console for each friend of the person with `name`.
    for friend in request.query("friend") {
        println!(
            "Hello, {}! Any friend of {} is a friend of mine.",
            name, friend
        );
    }

    // Send a response with our greeting message, repeated `n` times.
    Ok(message.repeat(n))
}
```